subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan_aux.c

```diff
@@ -723,7 +723,7 @@ static int isr_rx_pdu(struct lll_scan *lll, struct lll_scan_aux *lll_aux,
 		}
 
 		if (!lll_aux_to_use) {
-			return -ENOBUFS;
+			return -ECHILD;
```

Comment:
Can you please expand the commit message on why `ECHILD` is a better return code than `ENOBUFS`, or other return codes such as `ENOENT` or `ESRCH`?

Reply:
Updated commit message and also added comments in the code.

Reply:
Does it change anything in further execution of the ISRs? Or is it just a convenience change, to have some different return value to distinguish it from e.g. no more free node_rx objects?

Reply:
-ENOBUFS is consistently used as no free buffers in other places, and -ECANCELED is consistently for radio event close/done. Yes, -ECHILD is used to different that LLL scheduling, which is like a child process/task spawned from the primary channel PDU scanning, stops due to an aux context not being assigned. 

Reasons this can occurs, 1. ULL execution did not run in time before LLL ISR runs, or 2. no more free aux context available to be allocated.